tests/annotator/annotator.cpp

```diff
@@ -1911,42 +1984,21 @@ TEST(Annotator, hashChangesAndUpdates)
     auto model = parser->parseModel(modelStringUniqueIds);
     auto annotator = libcellml::Annotator::create();
 
-    EXPECT_FALSE(annotator->isBuilt());
-
-    annotator->buildModelIndex(model);
-    EXPECT_TRUE(annotator->isBuilt());
+    EXPECT_FALSE(annotator->hasModel());
 
-    model->setId("iHaveChangedSinceTheLastBuild");
-    EXPECT_FALSE(annotator->isBuilt());
+    // Model is set and the id map is not created.
+    annotator->setModel(model);
 
-    // Can set id automatically, but this does not trigger a rebuild.
-    EXPECT_EQ("b4da55", annotator->assignId(libcellml::Annotator::Type::COMPONENT, model->component(0)));
-    EXPECT_FALSE(annotator->isBuilt());
+    // Can set id manually, this does not trigger a rebuild of the id map.
+    model->setId("iHaveChanged");
 
-    // Shouldn't be able to retrieve anything since the build is out of date.
+    // The id map is built when a query is made.
     EXPECT_EQ(nullptr, annotator->model("model_1"));
```

Comment:
This test is a bit odd now?  The id has changed, so you should rather test that the new id can now be found (ie: that the index has successfully detected the change and been rebuilt)?

Reply:
Is this comment still relevant? @kerimoyle? @hsorby?